run demo on Windows

env:
```
Windows 10;
node: 10.5
yarn 1.7
react-admin: 2.1.2
```
use **Powershell**

steps: 
- clone
- cd react-admin
- yarn
- cd ./example/demo
- `(set REACT_APP_DATA_PROVIDER=rest) ; (yarn start)`

result: 
```
Failed to compile
./src/fakeServer/graphql.js
Module not found: Can't resolve 'data-generator' in 'E:\code\work\react-admin\examples\demo\src\fakeServer'
```

I tried use absolute path, bu it seems so many package use this way...

How do I run this demo right? 

PS: with ubuntu 18.04, it works fine.

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 npm-debug.log
 yarn-error.log
+lerna-debug.log
 node_modules
 lib
 es6

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,6 +19,7 @@ before_install:
     - source ~/.bashrc
     - sh -e /etc/init.d/xvfb start
     - sleep 3
+script: make test
 branches:
   only:
     - master

--- a/Makefile
+++ b/Makefile
@@ -92,15 +92,15 @@ test: build test-unit lint test-e2e ## launch all tests
 test-unit: ## launch unit tests
 	@if [ "$(CI)" != "true" ]; then \
 		echo "Running unit tests..."; \
-		NODE_ENV=test NODE_ICU_DATA=node_modules/full-icu ./node_modules/.bin/jest; \
+		yarn test-unit; \
 	fi
 	@if [ "$(CI)" = "true" ]; then \
 		echo "Running unit tests in CI..."; \
-		NODE_ENV=test NODE_ICU_DATA=node_modules/full-icu ./node_modules/.bin/jest --runInBand; \
+		yarn test-unit-ci; \
 	fi
 
 test-unit-watch: ## launch unit tests and watch for changes
-	@NODE_ENV=test NODE_ICU_DATA=node_modules/full-icu ./node_modules/.bin/jest --watch
+	yarn test-unit --watch
 
 test-e2e: ## launch end-to-end tests
 	@if [ "$(build)" != "false" ]; then \

--- a/Makefile
+++ b/Makefile
@@ -11,19 +11,19 @@ run: run-simple
 run-simple: ## run the simple example
 	@yarn run-simple
 
-run-tutorial: build ## run the tutorial example
+run-tutorial: ## run the tutorial example
 	@yarn run-tutorial
 
-run-demo: build ## run the demo example
+run-demo: ## run the demo example
 	@yarn run-demo
 
 build-demo: ## compile the demo example to static js
 	@yarn build-demo
 
-run-graphql-demo: build ## run the demo example
+run-graphql-demo: ## run the demo example
 	@yarn run-graphql-demo
 
-run-graphcool-demo: build ## run the demo example
+run-graphcool-demo: ## run the demo example
 	@yarn run-graphcool-demo
 
 build-ra-core:

--- a/Makefile
+++ b/Makefile
@@ -9,22 +9,22 @@ install: package.json ## install dependencies
 run: run-simple
 
 run-simple: ## run the simple example
-	@cd examples/simple && yarn start
+	@yarn run-simple
 
 run-tutorial: build ## run the tutorial example
-	@cd examples/tutorial && yarn start
+	@yarn run-tutorial
 
 run-demo: build ## run the demo example
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn start
+	@yarn run-demo
 
 build-demo: ## compile the demo example to static js
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=rest yarn build
+	@yarn build-demo
 
 run-graphql-demo: build ## run the demo example
-	@cd examples/demo && REACT_APP_DATA_PROVIDER=graphql yarn start
+	@yarn run-graphql-demo
 
 run-graphcool-demo: build ## run the demo example
-	@cd examples/graphcool-demo && yarn start
+	@yarn run-graphcool-demo
 
 build-ra-core:
 	@echo "Transpiling ra-core files...";

--- a/Makefile
+++ b/Makefile
@@ -28,61 +28,49 @@ run-graphcool-demo: build ## run the demo example
 
 build-ra-core:
 	@echo "Transpiling ra-core files...";
-	@rm -rf ./packages/ra-core/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-core/src -d ./packages/ra-core/lib --ignore spec.js,test.js
+	@cd ./packages/ra-core && yarn build
 
 build-ra-ui-materialui:
 	@echo "Transpiling ra-ui-materialui files...";
-	@rm -rf ./packages/ra-ui-materialui/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-ui-materialui/src -d ./packages/ra-ui-materialui/lib --ignore spec.js,test.js
+	@cd ./packages/ra-ui-materialui && yarn build
 
 build-react-admin:
 	@echo "Transpiling react-admin files...";
-	@rm -rf ./packages/react-admin/lib
-	@rm -rf ./packages/react-admin/docs
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/react-admin/src -d ./packages/react-admin/lib --ignore spec.js,test.js
+	@cd ./packages/react-admin && yarn build
 	@mkdir packages/react-admin/docs
 	@cp docs/*.md packages/react-admin/docs
 
 build-ra-data-json-server:
 	@echo "Transpiling ra-data-json-server files...";
-	@rm -rf ./packages/ra-data-json-server/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-data-json-server/src -d ./packages/ra-data-json-server/lib --ignore spec.js,test.js
+	@cd ./packages/ra-data-json-server && yarn build
 
 build-ra-data-simple-rest:
 	@echo "Transpiling ra-data-simple-rest files...";
-	@rm -rf ./packages/ra-data-simple-rest/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-data-simple-rest/src -d ./packages/ra-data-simple-rest/lib --ignore spec.js,test.js
+	@cd ./packages/ra-data-simple-rest && yarn build
 
 build-ra-data-graphql:
 	@echo "Transpiling ra-data-graphql files...";
-	@rm -rf ./packages/ra-data-graphql/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-data-graphql/src -d ./packages/ra-data-graphql/lib --ignore spec.js,test.js
+	@cd ./packages/ra-data-graphql && yarn build
 
 build-ra-data-graphcool:
 	@echo "Transpiling ra-data-graphcool files...";
-	@rm -rf ./packages/ra-data-graphcool/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-data-graphcool/src -d ./packages/ra-data-graphcool/lib --ignore spec.js,test.js
+	@cd ./packages/ra-data-graphcool && yarn build
 
 build-ra-data-graphql-simple:
 	@echo "Transpiling ra-data-graphql-simple files...";
-	@rm -rf ./packages/ra-data-graphql-simple/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-data-graphql-simple/src -d ./packages/ra-data-graphql-simple/lib --ignore spec.js,test.js
+	@cd ./packages/ra-data-graphql-simple && yarn build
 
 build-ra-input-rich-text:
 	@echo "Transpiling ra-input-rich-text files...";
-	@rm -rf ./packages/ra-input-rich-text/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-input-rich-text/src -d ./packages/ra-input-rich-text/lib --ignore spec.js,test.js
+	@cd ./packages/ra-input-rich-text && yarn build
 
 build-ra-realtime:
 	@echo "Transpiling ra-realtime files...";
-	@rm -rf ./packages/ra-realtime/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./packages/ra-realtime/src -d ./packages/ra-realtime/lib --ignore spec.js,test.js
+	@cd ./packages/ra-realtime && yarn build
 
 build-data-generator:
 	@echo "Transpiling data-generator files...";
-	@rm -rf ./examples/data-generator/lib
-	@NODE_ENV=production ./node_modules/.bin/babel --quiet ./examples/data-generator/src -d ./examples/data-generator/lib
+	@cd ./examples/data-generator && yarn build
 
 build: build-ra-core build-ra-ui-materialui build-react-admin build-ra-data-json-server build-ra-data-simple-rest build-ra-data-graphql build-ra-data-graphcool build-ra-data-graphql-simple build-ra-input-rich-text build-ra-realtime build-data-generator ## compile ES6 files to JS
 

--- a/Makefile
+++ b/Makefile
@@ -74,18 +74,16 @@ build-data-generator:
 
 build: build-ra-core build-ra-ui-materialui build-react-admin build-ra-data-json-server build-ra-data-simple-rest build-ra-data-graphql build-ra-data-graphcool build-ra-data-graphql-simple build-ra-input-rich-text build-ra-realtime build-data-generator ## compile ES6 files to JS
 
-watch: ## continuously compile ES6 files to JS
-	@NODE_ENV=production ./node_modules/.bin/babel ./src -d lib --ignore spec.js,test.js --watch
-
 doc: ## compile doc as html and launch doc web server
-	@cd docs && jekyll server . --watch
+	@yarn doc
 
 lint: ## lint the code and check coding conventions
 	@echo "Running linter..."
-	@"./node_modules/.bin/eslint" ./packages/**/src
+	@yarn lint
 
 prettier: ## prettify the source code using prettier
-	@./node_modules/.bin/prettier-eslint --write --list-different "packages/*/src/**/*.js" "examples/*/src/**/*.js"
+	@echo "Running prettier..."
+	@yarn prettier
 
 test: build test-unit lint test-e2e ## launch all tests
 

--- a/README.md
+++ b/README.md
@@ -157,6 +157,9 @@ You can run those example applications by calling:
 # At the react-admin project root
 make install
 
+# or
+yarn install
+
 # Run the simple application
 make run-simple
 # Run the tutorial application
@@ -177,7 +180,13 @@ You can run the tests (linting, unit and functional tests) by calling
 make test
 ```
 
-When working on the end to end tests, you can leverage [cypress](https://www.cypress.io/) runner by starting the simple example yourself (`make run-simple`) and starting cypress in another terminal (`make tests-e2e-local`).
+or
+
+```sh
+yarn test
+```
+
+When working on the end to end tests, you can leverage [cypress](https://www.cypress.io/) runner by starting the simple example yourself (`make run-simple`) and starting cypress in another terminal (`make test-e2e-local` or `yarn test-e2e-local`).
 
 If you have coding standards problems, you can fix them automatically using `prettier` by calling
 
@@ -185,10 +194,22 @@ If you have coding standards problems, you can fix them automatically using `pre
 make prettier
 ```
 
-If you want to contribute to the documentation, install jekyll, then call
+or
+
+```sh
+yarn prettier
+```
+
+If you want to contribute to the documentation, install [jekyll](https://jekyllrb.com/docs/home/), then call
 
 ```sh
 make doc
+```
+
+or
+
+```sh
+yarn doc
 ```
 
 And then browse to [http://localhost:4000/](http://localhost:4000/)

--- a/examples/data-generator/package.json
+++ b/examples/data-generator/package.json
@@ -3,10 +3,18 @@
     "version": "0.1.0",
     "private": true,
     "main": "./lib/index.js",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "babel-core": "~6.26.0",
         "chance": "~1.0.13",
         "date-fns": "~1.29.0",
         "md5": "~2.2.1"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
     }
 }

--- a/examples/data-generator/package.json
+++ b/examples/data-generator/package.json
@@ -4,7 +4,8 @@
     "private": true,
     "main": "./lib/index.js",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "babel-core": "~6.26.0",

--- a/lerna.json
+++ b/lerna.json
@@ -1,7 +1,8 @@
 {
   "lerna": "2.5.1",
   "packages": [
-    "packages/*"
+    "packages/*",
+    "examples/data-generator"
   ],
   "version": "2.1.2"
 }

--- a/package.json
+++ b/package.json
@@ -8,7 +8,10 @@
         "test-e2e": "npm run build && cross-env NODE_ENV=test && cd cypress && yarn test",
         "test-e2e-local": "cd cypress && yarn start",
         "test": "yarn test-unit && yarn test-e2e",
-        "test-ci": "yarn test-unit-ci && yarn test-e2e"
+        "test-ci": "yarn test-unit-ci && yarn test-e2e",
+        "doc": "cd docs && jekyll server . --watch",
+        "lint": "eslint \" ./packages /**/src\"",
+        "prettier": "prettier-eslint --write --list-different \"packages /*/src/**/*.js\" \"examples /*/src/**/*.js\""
     },
     "jest": {
         "setupTestFrameworkScriptFile": "./test-setup.js"
@@ -26,6 +29,7 @@
         "babel-preset-stage-0": "~6.24.1",
         "babel-register": "~6.26.0",
         "cheerio": "~1.0.0-rc.2",
+        "cross-env": "^5.2.0",
         "eslint": "~4.19.1",
         "eslint-config-prettier": "~2.9.0",
         "eslint-plugin-cypress": "~2.0.1",

--- a/package.json
+++ b/package.json
@@ -2,6 +2,7 @@
     "private": true,
     "name": "react-admin-lerna",
     "scripts": {
+        "build": "lerna run build",
         "test": "make test"
     },
     "jest": {

--- a/package.json
+++ b/package.json
@@ -16,8 +16,8 @@
         "run-simple": "cd examples/simple && yarn start",
         "run-tutorial": "yarn run build && cd examples/tutorial && yarn start",
         "run-demo": "yarn run build && cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=rest yarn start",
-        "run-demo-graphql": "yarn run build && cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=graphql yarn start",
-        "run-demo-graphcool": "yarn run build && cd examples/graphcool-demo && yarn start"
+        "run-graphql-demo": "yarn run build && cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=graphql yarn start",
+        "run-graphcool-demo": "yarn run build && cd examples/graphcool-demo && yarn start"
     },
     "jest": {
         "setupTestFrameworkScriptFile": "./test-setup.js"

--- a/package.json
+++ b/package.json
@@ -3,15 +3,21 @@
     "name": "react-admin-lerna",
     "scripts": {
         "build": "lerna run build",
+        "build-demo": "cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=rest yarn build",
         "test-unit": "cross-env NODE_ENV=test cross-env NODE_ICU_DATA=node_modules/full-icu jest",
         "test-unit-ci": "cross-env NODE_ENV=test cross-env NODE_ICU_DATA=node_modules/full-icu jest --runInBand",
-        "test-e2e": "npm run build && cross-env NODE_ENV=test && cd cypress && yarn test",
+        "test-e2e": "yarn run build && cross-env NODE_ENV=test && cd cypress && yarn test",
         "test-e2e-local": "cd cypress && yarn start",
         "test": "yarn test-unit && yarn test-e2e",
         "test-ci": "yarn test-unit-ci && yarn test-e2e",
         "doc": "cd docs && jekyll server . --watch",
         "lint": "eslint \" ./packages /**/src\"",
-        "prettier": "prettier-eslint --write --list-different \"packages /*/src/**/*.js\" \"examples /*/src/**/*.js\""
+        "prettier": "prettier-eslint --write --list-different \"packages /*/src/**/*.js\" \"examples /*/src/**/*.js\"",
+        "run-simple": "cd examples/simple && yarn start",
+        "run-tutorial": "yarn run build && cd examples/tutorial && yarn start",
+        "run-demo": "yarn run build && cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=rest yarn start",
+        "run-demo-graphql": "yarn run build && cd examples/demo && cross-env REACT_APP_DATA_PROVIDER=graphql yarn start",
+        "run-demo-graphcool": "yarn run build && cd examples/graphcool-demo && yarn start"
     },
     "jest": {
         "setupTestFrameworkScriptFile": "./test-setup.js"

--- a/package.json
+++ b/package.json
@@ -3,7 +3,12 @@
     "name": "react-admin-lerna",
     "scripts": {
         "build": "lerna run build",
-        "test": "make test"
+        "test-unit": "cross-env NODE_ENV=test cross-env NODE_ICU_DATA=node_modules/full-icu jest",
+        "test-unit-ci": "cross-env NODE_ENV=test cross-env NODE_ICU_DATA=node_modules/full-icu jest --runInBand",
+        "test-e2e": "npm run build && cross-env NODE_ENV=test && cd cypress && yarn test",
+        "test-e2e-local": "cd cypress && yarn start",
+        "test": "yarn test-unit && yarn test-e2e",
+        "test-ci": "yarn test-unit-ci && yarn test-e2e"
     },
     "jest": {
         "setupTestFrameworkScriptFile": "./test-setup.js"

--- a/packages/ra-core/package.json
+++ b/packages/ra-core/package.json
@@ -16,13 +16,19 @@
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "enzyme": "~3.3.0",
         "enzyme-adapter-react-16": "~1.1.1",
         "ignore-styles": "~5.0.1",
         "react": "~16.3.1",
         "react-dom": "~16.3.1",
-        "react-test-renderer": "~16.3.1"
+        "react-test-renderer": "~16.3.1",
+        "rimraf": "^2.6.2"
     },
     "peerDependencies": {
         "react": "~16.0.0 || ~16.1.0 || ~16.2.0 || ~16.3.0",

--- a/packages/ra-core/package.json
+++ b/packages/ra-core/package.json
@@ -17,7 +17,8 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "devDependencies": {
         "babel-cli": "~6.26.0",

--- a/packages/ra-data-fakerest/package.json
+++ b/packages/ra-data-fakerest/package.json
@@ -28,7 +28,8 @@
     },
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "fakerest": "~2.1.0"

--- a/packages/ra-data-fakerest/package.json
+++ b/packages/ra-data-fakerest/package.json
@@ -3,9 +3,6 @@
     "version": "2.0.2",
     "description": "JSON Server data provider for react-admin",
     "main": "lib/index.js",
-    "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
-    },
     "files": [
         "LICENSE",
         "*.md",
@@ -30,8 +27,16 @@
         "url": "https://github.com/marmelab/react-admin/issues"
     },
     "homepage": "https://github.com/marmelab/react-admin#readme",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "fakerest": "~2.1.0"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
     },
     "peerDependencies": {
         "react-admin": "^2.0.0"

--- a/packages/ra-data-graphcool/package.json
+++ b/packages/ra-data-graphcool/package.json
@@ -25,7 +25,8 @@
     ],
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "graphql-tag": "^2.6.1",

--- a/packages/ra-data-graphcool/package.json
+++ b/packages/ra-data-graphcool/package.json
@@ -24,6 +24,9 @@
         "graphcool"
     ],
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "graphql-tag": "^2.6.1",
         "lodash": "~4.17.5",
@@ -35,7 +38,10 @@
         "react-admin": "^2.1.2"
     },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "graphql": "^0.13.2",
-        "react-admin": "^2.1.2"
+        "react-admin": "^2.1.2",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-data-graphql-simple/package.json
+++ b/packages/ra-data-graphql-simple/package.json
@@ -23,6 +23,9 @@
         "graphql"
     ],
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "graphql-tag": "^2.6.1",
         "lodash": "~4.17.5",
@@ -34,7 +37,10 @@
         "react-admin": "^2.1.2"
     },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "graphql": "^0.13.2",
-        "react-admin": "^2.1.2"
+        "react-admin": "^2.1.2",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-data-graphql-simple/package.json
+++ b/packages/ra-data-graphql-simple/package.json
@@ -24,7 +24,8 @@
     ],
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "graphql-tag": "^2.6.1",

--- a/packages/ra-data-graphql/package.json
+++ b/packages/ra-data-graphql/package.json
@@ -24,7 +24,8 @@
     ],
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "apollo-client": "^2.2.8",

--- a/packages/ra-data-graphql/package.json
+++ b/packages/ra-data-graphql/package.json
@@ -23,6 +23,9 @@
         "graphql"
     ],
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "apollo-client": "^2.2.8",
         "apollo-client-preset": "^1.0.6",
@@ -35,7 +38,10 @@
         "react-admin": "^2.1.2"
     },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "graphql": "^0.13.2",
-        "react-admin": "^2.1.2"
+        "react-admin": "^2.1.2",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-data-json-server/package.json
+++ b/packages/ra-data-json-server/package.json
@@ -16,7 +16,8 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "query-string": "~5.1.1",

--- a/packages/ra-data-json-server/package.json
+++ b/packages/ra-data-json-server/package.json
@@ -8,9 +8,6 @@
         "lib",
         "src"
     ],
-    "scripts": {
-        "test": "make test"
-    },
     "authors": [
         "François Zaninotto"
     ],
@@ -18,8 +15,16 @@
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "query-string": "~5.1.1",
         "react-admin": "^2.1.2"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-data-simple-rest/package.json
+++ b/packages/ra-data-simple-rest/package.json
@@ -16,7 +16,8 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "query-string": "~5.1.1",

--- a/packages/ra-data-simple-rest/package.json
+++ b/packages/ra-data-simple-rest/package.json
@@ -8,9 +8,6 @@
         "lib",
         "src"
     ],
-    "scripts": {
-        "test": "make test"
-    },
     "authors": [
         "François Zaninotto"
     ],
@@ -18,8 +15,16 @@
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "query-string": "~5.1.1",
         "react-admin": "^2.1.2"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-input-rich-text/package.json
+++ b/packages/ra-input-rich-text/package.json
@@ -31,7 +31,8 @@
     },
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "peerDependencies": {
         "react": "~16.0.0 || ~16.1.0 || ~16.2.0 || ~16.3.0",

--- a/packages/ra-input-rich-text/package.json
+++ b/packages/ra-input-rich-text/package.json
@@ -3,9 +3,6 @@
     "version": "2.1.0",
     "description": "<RichTextInput> component for react-admin, useful for editing HTML code in admin GUIs.",
     "main": "lib/index.js",
-    "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
-    },
     "files": [
         "LICENSE",
         "*.md",
@@ -33,6 +30,9 @@
         "url": "https://github.com/marmelab/react-admin/issues"
     },
     "homepage": "https://github.com/marmelab/react-admin#readme",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "peerDependencies": {
         "react": "~16.0.0 || ~16.1.0 || ~16.2.0 || ~16.3.0",
         "react-admin": "^2.0.0",
@@ -42,5 +42,10 @@
         "lodash": "~4.17.5",
         "prop-types": "~15.6.1",
         "quill": "~1.3.6"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-language-english/package.json
+++ b/packages/ra-language-english/package.json
@@ -3,9 +3,6 @@
     "version": "2.0.2",
     "description": "English messages for react-admin, the frontend framework for building admin applications on top of REST/GraphQL services",
     "main": "index.js",
-    "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
-    },
     "repository": {
         "type": "git",
         "url": "git+https://github.com/marmelab/react-admin.git"

--- a/packages/ra-language-french/package.json
+++ b/packages/ra-language-french/package.json
@@ -3,9 +3,6 @@
     "version": "2.0.2",
     "description": "French messages for react-admin, the frontend framework for building admin applications on top of REST/GraphQL services",
     "main": "index.js",
-    "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1"
-    },
     "repository": {
         "type": "git",
         "url": "git+https://github.com/marmelab/react-admin.git"

--- a/packages/ra-realtime/package.json
+++ b/packages/ra-realtime/package.json
@@ -24,7 +24,8 @@
     ],
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "dependencies": {
         "lodash": "~4.17.5"

--- a/packages/ra-realtime/package.json
+++ b/packages/ra-realtime/package.json
@@ -23,6 +23,9 @@
         "realtime"
     ],
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "dependencies": {
         "lodash": "~4.17.5"
     },
@@ -33,9 +36,12 @@
         "redux-saga": "~0.16.0"
     },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "react-admin": "^2.1.2",
         "react-router": "~4.2.0",
         "react-router-redux": "~5.0.0-alpha.9",
-        "redux-saga": "~0.16.0"
+        "redux-saga": "~0.16.0",
+        "rimraf": "^2.6.2"
     }
 }

--- a/packages/ra-ui-materialui/package.json
+++ b/packages/ra-ui-materialui/package.json
@@ -16,14 +16,20 @@
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
     "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
         "enzyme": "~3.3.0",
         "enzyme-adapter-react-16": "~1.1.1",
         "file-api": "~0.10.4",
         "ignore-styles": "~5.0.1",
         "react": "~16.3.1",
         "react-dom": "~16.3.1",
-        "react-test-renderer": "~16.3.1"
+        "react-test-renderer": "~16.3.1",
+        "rimraf": "^2.6.2"
     },
     "peerDependencies": {
         "react": "~16.3.0",

--- a/packages/ra-ui-materialui/package.json
+++ b/packages/ra-ui-materialui/package.json
@@ -17,7 +17,8 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "devDependencies": {
         "babel-cli": "~6.26.0",

--- a/packages/react-admin/package.json
+++ b/packages/react-admin/package.json
@@ -16,6 +16,14 @@
     "homepage": "https://github.com/marmelab/react-admin#readme",
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
+    "scripts": {
+        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+    },
+    "devDependencies": {
+        "babel-cli": "~6.26.0",
+        "cross-env": "^5.2.0",
+        "rimraf": "^2.6.2"
+    },
     "peerDependencies": {
         "react": "~16.3.0",
         "react-dom": "~16.3.0"

--- a/packages/react-admin/package.json
+++ b/packages/react-admin/package.json
@@ -17,7 +17,8 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "rimraf ./lib && NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js"
+        "build": "rimraf ./lib && cross-env NODE_ENV=production babel --quiet ./src -d ./lib --ignore spec.js,test.js",
+        "watch": "rimraf ./lib && cross-env NODE_ENV=production babel --watch ./src -d ./lib --ignore spec.js,test.js"
     },
     "devDependencies": {
         "babel-cli": "~6.26.0",

